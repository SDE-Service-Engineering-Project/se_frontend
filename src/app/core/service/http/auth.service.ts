import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {AuthResponse} from "../../model/AuthResponse";

const AUTH_API = 'http://localhost:8080/auth/';

@Injectable({
  providedIn: 'root'
})
export class AuthService {

  constructor(private http: HttpClient) {
  }

  public login(userName: string, password: string): Observable<AuthResponse> {
    return this.http.post<AuthResponse>(
      AUTH_API + 'login',
      {userName, password}
    );
  }
}
