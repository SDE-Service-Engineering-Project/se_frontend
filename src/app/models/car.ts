export interface Car {
  carId: number;
  createdOn: string;
  brand: string;
  model: string;
  constructionYear: number;
  price: number;
  precision: number;
  currency: string;
}
