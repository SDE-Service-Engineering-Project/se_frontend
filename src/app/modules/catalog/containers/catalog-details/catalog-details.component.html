<div class="details container">
  <div class="details__title row">
    <h1>{{ car.model }} {{ car.brand }}</h1>
  </div>
  <div class="row row-cols-auto details__row-1">
    <div class="col details__row-1__image">
      <img
        src="https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Ftse1.mm.bing.net%2Fth%3Fid%3DOIP.GDp9uheqSrMQBmNhBUP1mwHaE0%26pid%3DApi&f=1&ipt=ee353c7a978b7ca256f57cae6ae51c4d5ca321133e806df55516214bfb5ba431&ipo=images"
        class="card-img-top"
        alt="{{ car.brand }} {{ car.model }}"
      />
    </div>
    <div class="col details__row-1__date-picker">
      <div class="row">
        <app-date-picker
          (startDate)="setStartDate($event)"
          (endDate)="setEndDate($event)"
        ></app-date-picker>
      </div>
      <div class="row details__row-1__day-price">
        <div class="col">Total Per Day:</div>
        <div class="col">{{ price }} {{ currencyService.getCurrentCurrency() }}</div>
      </div>
      <div class="row details__row-1__separator">
        <hr />
      </div>
      <div class="row details__row-1__total-price">
        <div class="col">Total:</div>
        <div style="display: flex" class="col">
          <div style="margin-right: 4px">
            {{ getTotalPrice() }}
          </div>
          <div>
            <div class="nav-item" ngbDropdown>
              <a
                class="nav-link"
                tabindex="0"
                ngbDropdownToggle
                id="navbarDropdown1"
                role="button"
              >
                {{ currencyService.getCurrentCurrency() }}
              </a>
              <div
                *ngIf="currencies$ | async as currencies"
                ngbDropdownMenu
                aria-labelledby="navbarDropdown1"
                class="dropdown-menu"
              >
                <p
                  ngbDropdownItem
                  *ngFor="let currency of currencies.currencies"
                  (click)="setCurrency(currency)"
                >
                  {{ currency }}
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="row details__row-1__rsv-btn">
        <button
          class="btn btn-primary"
          data-testid="book-car-btn"
          [disabled]="!checkIfDatesAreValid()"
          (click)="bookCar()"
        >
          Book Car
        </button>
      </div>
    </div>
  </div>
  <div class="row details__row-2">
    <div class="col">
      <div class="row row-cols-2">
        <div class="col">
          <p class="card-text details__row-2__car-properties">Brand:</p>
        </div>
        <div class="col">
          <p class="card-text details__row-2__car-properties">
            {{ car.brand }}
          </p>
        </div>
      </div>
      <div class="row row-cols-2">
        <div class="col">
          <p class="card-text details__row-2__car-properties">Model:</p>
        </div>
        <div class="col">
          <p class="card-text details__row-2__car-properties">
            {{ car.model }}
          </p>
        </div>
      </div>
      <div class="row row-cols-2">
        <div class="col">
          <p class="card-text details__row-2__car-properties">
            Manufacture Year:
          </p>
        </div>
        <div class="col">
          <p class="card-text details__row-2__car-properties">
            {{ car.constructionYear }}
          </p>
        </div>
      </div>
      <div class="row row-cols-2">
        <div class="col">
          <p class="card-text details__row-2__car-properties">Price per day:</p>
        </div>
        <div class="col">
          <p class="card-text details__row-2__car-properties">
            {{  price }} {{ currencyService.getCurrentCurrency() }}
          </p>
        </div>
      </div>
    </div>
    <div class="col"></div>
  </div>
</div>
