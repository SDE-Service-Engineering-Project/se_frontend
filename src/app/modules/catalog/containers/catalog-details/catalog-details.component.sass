.details
  &__title
    margin: 2rem

  &__row-1
    &__day-price
      margin: 1.5rem 0
      font-size: medium


    &__total-price
      margin: 1rem 0
      font-weight: bold
      font-size: large


  &__row-2
    &__car-properties
      font-size: x-large

      &__details
        margin-left: 2rem
