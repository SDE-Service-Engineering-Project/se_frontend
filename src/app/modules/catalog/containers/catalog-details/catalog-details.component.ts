import { Component } from '@angular/core';

@Component({
  selector: 'app-catalog-details',
  templateUrl: './catalog-details.component.html',
  styleUrls: ['./catalog-details.component.sass']
})
export class CatalogDetailsComponent {

}
