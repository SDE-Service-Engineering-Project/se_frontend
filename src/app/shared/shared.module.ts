import { NgModule } from '@angular/core';
import { HeaderComponent } from './components/header/header.component';
import { RouterLink, RouterLinkActive } from '@angular/router';
import { CommonModule } from '@angular/common';
import { NgbCollapse, NgbToast } from '@ng-bootstrap/ng-bootstrap';
import { ToastComponent } from './components/toast/toast.component';
import { NotFoundComponent } from './components/not-found/not-found.component';
import { MapsComponent } from './components/maps/maps.component';
import { GoogleMapsModule } from '@angular/google-maps';

@NgModule({
  declarations: [
    HeaderComponent,
    ToastComponent,
    NotFoundComponent,
    MapsComponent,
  ],
  exports: [HeaderComponent, ToastComponent, MapsComponent],
  imports: [
    RouterLink,
    CommonModule,
    RouterLinkActive,
    NgbCollapse,
    NgbToast,
    GoogleMapsModule,
  ],
})
export class SharedModule {}
