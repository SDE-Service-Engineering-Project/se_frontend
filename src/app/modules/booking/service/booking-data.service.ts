import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Booking } from '../../../models/booking';

export const BOOKING_API = 'http://localhost:8080/api/v1/bookings';

@Injectable({
  providedIn: 'root',
})
export class BookingDataService {
  constructor(private http: HttpClient) {}

  fetchBookings(): Observable<Booking[]> {
    return this.http.get<Booking[]>(BOOKING_API);
  }
}
