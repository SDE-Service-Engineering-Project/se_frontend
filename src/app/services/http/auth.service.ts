import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {AuthResponse} from '../../models/AuthResponse';

export const AUTH_API = 'http://localhost:8080/api/v1/auth/';

@Injectable({
  providedIn: 'root',
})
export class AuthService {
  constructor(private http: HttpClient) {
  }

  public login(userName: string, password: string): Observable<AuthResponse> {
    return this.http.post<AuthResponse>(AUTH_API + 'login', {
      userName,
      password,
    });
  }

  public register(userName: string, firstName: string, lastName: string, password: string): Observable<any> {
    return this.http.post(AUTH_API + 'signup', {
      userName,
      firstName,
      lastName,
      password,
    });
  }

  public refreshToken(refreshToken: string, userName: string): Observable<AuthResponse> {
    return this.http.post<AuthResponse>(AUTH_API + 'refresh/token', {
      refreshToken,
      userName,
    });
  }
}
