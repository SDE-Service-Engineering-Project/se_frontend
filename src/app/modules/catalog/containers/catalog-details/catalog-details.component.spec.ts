import { ComponentFixture } from '@angular/core/testing';

import { CatalogDetailsComponent } from './catalog-details.component';
import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
import { NgbInputDatepicker } from '@ng-bootstrap/ng-bootstrap';
import { SharedModule } from '../../../../shared/shared.module';
import { HttpClientTestingModule } from '@angular/common/http/testing';
import { ActivatedRoute, RouterModule } from '@angular/router';
import { carMock1 } from '../../../../utils/testing/mocks/car.mock';
import { CarDataService } from '../../services/car-data.service';

describe('CatalogDetailsComponent', () => {
  let component: CatalogDetailsComponent;
  let fixture: ComponentFixture<CatalogDetailsComponent>;
  let spectator: Spectator<CatalogDetailsComponent>;
  let activatedRoute: ActivatedRoute;

  const createComponent = createComponentFactory({
    component: CatalogDetailsComponent,
    imports: [
      NgbInputDatepicker,
      SharedModule,
      RouterModule,
      HttpClientTestingModule,
    ],
    providers: [
      CarDataService,
      {
        provide: ActivatedRoute,
        useValue: {
          snapshot: {
            params: {
              carId: carMock1.carId,
            },
          },
        },
      },
    ],
  });

  beforeEach(() => {
    spectator = createComponent();
    component = spectator.component;
    fixture = spectator.fixture;
    activatedRoute = spectator.inject(ActivatedRoute);
  });

  it('should render the component', () => {
    expect(fixture.debugElement.nativeElement).toMatchSnapshot();
  });
});
