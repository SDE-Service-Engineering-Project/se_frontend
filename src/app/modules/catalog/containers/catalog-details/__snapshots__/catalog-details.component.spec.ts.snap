// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogDetailsComponent should render the component 1`] = `
<div
  id="root0"
>
  <div
    class="details container"
  >
    <div
      class="details__title row"
    >
      <h1>
         
      </h1>
    </div>
    <div
      class="row details__row-1"
    >
      <div
        class="col details__row-1__image"
      >
        <img
          alt=" "
          class="card-img-top"
          src="https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Ftse1.mm.bing.net%2Fth%3Fid%3DOIP.GDp9uheqSrMQBmNhBUP1mwHaE0%26pid%3DApi&f=1&ipt=ee353c7a978b7ca256f57cae6ae51c4d5ca321133e806df55516214bfb5ba431&ipo=images"
        />
      </div>
      <div
        class="col details__row-1__date-picker"
      >
        <div
          class="row"
        >
          <app-date-picker>
            <form
              class="row row-cols-sm-auto ng-untouched ng-pristine ng-valid"
              novalidate=""
            >
              <div
                class="col-12"
              >
                <label
                  class="form-label"
                  for="pick-up-date"
                >
                  Pick-up Date
                </label>
                <div
                  class="dp-hidden position-absolute"
                >
                  <div
                    class="input-group"
                  >
                    <input
                      class="form-control"
                      name="datepicker"
                      ngbdatepicker=""
                      outsidedays="hidden"
                      tabindex="-1"
                    />
                    
                    
                  </div>
                </div>
                <div
                  class="input-group"
                >
                  <input
                    class="form-control"
                    id="pick-up-date"
                    name="dpFromDate"
                    placeholder="yyyy-mm-dd"
                  />
                  <button
                    class="btn btn-outline-secondary bi bi-calendar3"
                    data-testid="from-date-btn"
                    type="button"
                  />
                </div>
              </div>
              <div
                class="col-12"
              >
                <label
                  class="form-label"
                  for="return-date"
                >
                  Return Date
                </label>
                <div
                  class="input-group"
                >
                  <input
                    class="form-control"
                    id="return-date"
                    name="dpToDate"
                    placeholder="yyyy-mm-dd"
                  />
                  <button
                    class="btn btn-outline-secondary bi bi-calendar3"
                    data-testid="to-date-btn"
                    type="button"
                  />
                </div>
              </div>
            </form>
          </app-date-picker>
        </div>
        <div
          class="row details__row-1__day-price"
        >
          <div
            class="col"
          >
            Total Per Day:
          </div>
          <div
            class="col"
          >
             
          </div>
        </div>
        <div
          class="row details__row-1__separator"
        >
          <hr />
        </div>
        <div
          class="row details__row-1__total-price"
        >
          <div
            class="col"
          >
            Total:
          </div>
          <div
            class="col"
          >
             10
          </div>
        </div>
        <div
          class="row details__row-1__rsv-btn"
        >
          <button
            class="btn btn-primary"
            disabled=""
          >
            Reserve Car
          </button>
        </div>
      </div>
    </div>
    <div
      class="row details__row-2"
    >
      <div
        class="row"
      >
        <div
          class="col-sm-2"
        >
          <p
            class="card-text details__row-2__car-properties"
          >
            Brand:
          </p>
        </div>
        <div
          class="col-sm-2"
        >
          <p
            class="card-text details__row-2__car-properties"
          >
            
          </p>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-sm-2"
        >
          <p
            class="card-text details__row-2__car-properties"
          >
            Model:
          </p>
        </div>
        <div
          class="col-sm-2"
        >
          <p
            class="card-text details__row-2__car-properties"
          >
            
          </p>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-sm-2"
        >
          <p
            class="card-text details__row-2__car-properties"
          >
            Manufacture Year:
          </p>
        </div>
        <div
          class="col-sm-2"
        >
          <p
            class="card-text details__row-2__car-properties"
          >
            
          </p>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-sm-2"
        >
          <p
            class="card-text details__row-2__car-properties"
          >
            Price per day:
          </p>
        </div>
        <div
          class="col-sm-2"
        >
          <p
            class="card-text details__row-2__car-properties"
          >
            
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
